### PR TITLE
GUI: Port to Qt

### PR DESCRIPTION
**Pull request by michaelpm54**
Updates:
- Split pthread commit
- Fixed ax_check_qt5.m4 indentation
- Added verdigris license and readme
- Changes to dists/debian/copyright
- Changes to authors
- Split c++ standard commit
- Fixed improper usage of Doxygen comments /** Model functions. */ like this, now looks like // Model functions
- Removed newline in Doxygen comments

### CODE REVIEW

File: verdigris/rules.mk [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#
+# You should have received a copy of the GNU General Public License
+# along with Phaethon. If not, see <http://www.gnu.org/licenses/>.
+
+# Verdigris so we don't need moc (<https://github.com/woboq/verdigris>).
+
+noinst_HEADERS += \
+    verdigris/wobjectdefs.h \
+    verdigris/wobjectimpl.h \
+    $(EMPTY)
```
Comment (DrMcCoy) [inline verdigris/rules.mk]: Can you also add a README.Phaethon file into the verdigris directory, with the contents
```
This is the source of Verdigris (<https://github.com/woboq/verdigris>),
based on the git revision .
```
(and the revision you based it on filled in there).

Also, add a section for it in the AUTHORS file (probably right after UTF-8 CPP):
```
Verdigris
*********
   Phaethon uses and includes the header-only library Verdigris
   (<https://github.com/woboq/verdigris>) to use Qt5 without moc. Verdigris
   is licensed under version 3 or later of the Lesser GNU General Public
   License. For the full license texts, see LICENSE.LGPLv3 or
   <https://gnu.org/licenses/lgpl-3.0.html>.
```

And drop in that license file in plaintext.

Likewise, add the two Verdigris files to dists/debian/copyright:
```
Files: verdigris/wobjectdefs.h
       verdigris/wobjectimpl.h
Copyright: 2016-2018 Woboq GmbH 
License: LGPL-3.0+
```
And you then need a section
```
License: LGPL-3.0+
 This package is free software; you can redistribute it and/or
 modify it under the terms of the GNU Lesser General Public
 License as published by the Free Software Foundation; either
 version 3 of the License, or (at your option) any later version.
 .
 This package is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 Lesser General Public License for more details.
 .
 On Debian systems, the complete text of the GNU Library General Public License
 can be found in the file "/usr/share/common-licenses/LGPL-3".

```
at the bottom of the copyright file.

This should all go within this commit, please.

File: configure.ac [hunk 2]
```diff
@@ -132,11 +132,11 @@ fi
 AC_SUBST(WARN)
 
 dnl C++ dialect
-AC_ARG_WITH([std], [AS_HELP_STRING([--with-std], [Compile with specifying the C++ standard to follow @<:@default=no@:>@])], [], [with_std=no])
+AC_ARG_WITH([std], [AS_HELP_STRING([--without-std], [Compile without specifying the C++ standard to follow @<:@default=no@:>@])], [], [with_std=yes])
 
 STD=""
 if test "x$with_std" = "xyes"; then
-	AX_CHECK_COMPILER_FLAGS_VAR([C++], [STD], [-std=c++03])
+	AX_CHECK_COMPILER_FLAGS_VAR([C++], [STD], [-std=c++14])
```
Comment (DrMcCoy) [inline configure.ac]: Pushing the C++ standard to C++14 should be it own commit, please.

File: src/gui/resourcetree.h [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+	Aurora::Archive     *getArchive(const QString &path);
+	Aurora::KEYDataFile *getKEYDataFile(const QString &file);
+	void                loadKEYDataFiles(Aurora::KEYFile &key);
+
+	/** Return the item in the tree structure that corresponds to the given index.
+	*/
+	ResourceTreeItem *itemFromIndex(const QModelIndex &index) const;
+
+	/** Model functions. */
```
Comment (DrMcCoy) [inline src/gui/resourcetree.h]: Comments to denote a "section" shouldn't be Doxygen comments (i.e. please use `/*` here instead of `/**`)

This applies to resourcetreeitem.h as well and further commits as well.

File: src/gui/resourcetree.h [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+
+	/** Return the index if it exists, else create it.
+	*/
```
Comment (DrMcCoy) [inline src/gui/resourcetree.h]: Please remove these line breaks inside the Doxygen comments here. There's no need to make the comment span two lines.

This applies to resourcetreeitem.h as well and further commits as well.

**Comment by DrMcCoy** [comment]
This looks quite good now, yes. I pushed it to the temporary https://github.com/xoreos/phaethon/tree/qttravis branch to see what Travis CI says. :)

How do you want to be credited in the AUTHORS file now? And should I also change your line in the xoreos AUTHORS file?

**Comment by michaelpm54** [comment]
`Michael McAssey <michael.mcassey1@gmail.com>`

And yes, please do change the one in xoreos.

**Comment by DrMcCoy** [comment]
Merged as 2dc7e82...7d5b853, thanks! :D

**Comment by DrMcCoy** [comment]
Hmm, is there a reason for the dependency on QtMultimedia? It doesn't seem to be needed, it seems. I mean, we do have our own sound mixer, powered by OpenAl, after all.

I'm currently exploring potentially creating release packages of Phaethon. For Linux, I'm using a Debian chroot (Debian Stretch, for Phaethon). The thing is, the QtMultimedia in Debian pulls in pulseaudio, which pulls in systemd... which is not something I can distribute.

So I'd need to compile Qt myself in the chroot. I'd rather not, honestly. And if that can be prevented by not requiring QtMultimedia, that would make me happy.

**Comment by michaelpm54** [comment]
I think it's a relic from a previous iteration of phaethon-qt, in which I used it for sound and/or images.
I removed it and everything seems to be working fine.
